be/src/exprs/vectorized/string_functions.cpp

```diff
@@ -2052,6 +2053,41 @@ ColumnPtr StringFunctions::unhex(FunctionContext* context, const starrocks::vect
     return VectorizedStringStrictUnaryFunction<unhexImpl>::evaluate<TYPE_VARCHAR, TYPE_VARCHAR>(columns[0]);
 }
 
+DEFINE_STRING_UNARY_FN_WITH_IMPL(sm3Impl, str) {
+    const Slice& input_str = str;
+    std::stringstream result;
+
+    const unsigned char* message = (unsigned char*)input_str.data;
+    unsigned long message_len = input_str.size;
+    if (message_len > 0) {
+        unsigned char output[32];
+
+        // output as 256 bits(32 bytes) result.
+        Sm3::sm3_compute(message, message_len, output);
+        result << std::hex << std::setfill('0');
+
+        // first 4 bytes;
+        result << std::setw(2) << (output[0] & 0xFF);
```

Comment:
could this piece code be integrated into the following loop?

```
if ((i % 4) == 0 && i != 0) {
}
```

Reply:
ok